executor/executor_join.go

```diff
@@ -724,18 +729,17 @@ func (e *HashSemiJoinExec) Next() (*Row, error) {
 }
 
 // ApplyJoinExec is the new logic of apply.
-// TODO: I will add test for it after refactoring the plan part of apply.
 type ApplyJoinExec struct {
 	join        joinExec
-	innerExec   Executor
 	outerSchema []*expression.CorrelatedColumn
 	cursor      int
 	resultRows  []*Row
+	schema      expression.Schema
 }
 
 // Schema implements the Executor interface.
 func (e *ApplyJoinExec) Schema() expression.Schema {
-	return e.join.Schema()
+	return e.schema
```

Comment:
not exactly. because physical apply contains a join pointer. During projection elimination, projection schema will assignt to apply but join.